Documentation/Viper RxModule Guide.md

```diff
@@ -0,0 +1,560 @@
+
+    init(userServiceable: UserServiceable = RxUserService.shared) {
+        self.userServiceable = userServiceable
+    }
+}
+
+// MARK: - Extensions -
+
+extension RxLoginInteractor: RxLoginInteractorInterface {
+    func login(with email: String, _ password: String) -> Single<User> {
+        userServiceable.login(with: email, password)
+    }
+
+}
+```
+
+The interactor, as in the base module, helps us make API calls or any other call that we need. It contains a protocol as a property that has to know how to handle our needed function. In this case, our *UserServiceable* knows how to log in a user and that is its main purpose. If we had the register functionality on the login screen, we'd add register to the *UserServiceable* protocol. The main reason why it's a protocol that a class has to implement is that it's much easier to test ( create mocks ) and even if it has to login in another way the Interactor doesn't need to know how you do it, just that you do it.
+
+Last but not the least, *RxLoginWireframe*:
+
+### RxLoginWireframe
+
+```swift
+final class RxLoginWireframe: BaseWireframe {
+
+    // MARK: - Private properties -
+    private let storyboard = UIStoryboard(name: "RxLogin", bundle: nil)
+    // MARK: - Module setup -
+
+    init() {
+        let moduleViewController = storyboard.instantiateViewController(ofType: RxLoginViewController.self)
+        super.init(viewController: moduleViewController)
+
+        let interactor = RxLoginInteractor()
+        let presenter = RxLoginPresenter(view: moduleViewController, interactor: interactor, wireframe: self)
+        moduleViewController.presenter = presenter
+    }
+
+}
+
+// MARK: - Extensions -
+
+extension RxLoginWireframe: RxLoginWireframeInterface {
+    func navigate(to option: LoginNavigationOption) {
+        switch option {
+        case .home:
+            _openHome()
+        case .register:
+            _presentRegister()
+        }
+
+    }
+
+    private func _openHome() {
+        let wireframe = HomeWireframe()
+
+        navigationController?.pushWireframe(wireframe)
+    }
+
+    private func _presentRegister() {
+        let wireframe = RegisterWireframe()
+
+        let wireframeNavigationController = PokedexNavigationController()
+        wireframeNavigationController.setRootWireframe(wireframe)
+
+        navigationController?.present(wireframeNavigationController, animated: true, completion: nil)
+
+    }
+}
+```
+
+There isn't anything special going on here, that is specific for the RxModule. It's just handling the navigation from the Login to home/register.
+
+This here just shows you how you can implement on a simple example, it can get much more complicated than this by using a Formatter or just a bigger Presenter and ViewController. The sole purpose is to show you the tools and you can better yourself by using them. Hopefully, this wasn't too hard for you, and that you managed to get a clearer picture in your head of how the *RxModule* works.
```

Comment:
```suggestion
This here just shows you how you can implement on a simple example, it can get much more complicated than this by using a Formatter or just a bigger Presenter and ViewController. The sole purpose of this example is to show you the tools and how you can prospect when using them. Hopefully, this wasn't too hard for you, and that you managed to get a clearer picture in your head of how the *RxModule* works.
```